docs/developer/solutions/introduction.rst

```diff
@@ -0,0 +1,115 @@
+party in this integration, and will link to relevant documentation pages for
+detailed information.
+
+What is a *Solution*?
+---------------------
+
+We use the term *Solution* to describe a packaged Kubernetes application,
+archived as an ISO disk image, containing:
+
+- A set of OCI images to inject in MetalK8s image registry
+- An `Operator`_, to deploy on the cluster
+- Optionally, a UI for managing and monitoring the application, represented by
+  a standard Kubernetes ``Deployment``
+
+.. _Operator: https://coreos.com/blog/introducing-operators.html
+
+For more details, see the following documentation pages:
+
+- :doc:`./archive`
+- :doc:`./operator`
+- (TODO) Solution UI guidelines
+
+Once a Solution is deployed on MetalK8s, a user can deploy one or more versions
+of the Solution Operator, using either the Solution UI or the Kubernetes API,
+into separate namespaces. Using the Operator-defined ``CustomResource(s)``, the
+user can then effectively deploy the application packaged in the Solution.
+
+How is a *Solution* declared in MetalK8s?
+-----------------------------------------
+
+MetalK8s already uses a ``BootstrapConfiguration`` object, stored in
+``/etc/metalk8s/bootstrap.yaml``, to define how the cluster should be
+configured from the bootstrap node, and what versions of MetalK8s are available
+to the cluster.
+
+In the same vein, we want to use a ``SolutionsConfiguration`` object, stored in
+``/etc/metalk8s/solutions.yaml``, to declare which Solutions are available to
+the cluster, from the bootstrap node.
+
+.. todo:: Add specification in a future Reference guide
+
+Here is how it could look::
+
+    apiVersion: metalk8s.scality.com/v1alpha1
+    kind: SolutionsConfiguration
+    solutions:
+      - /solutions/storage_1.0.0.iso
+      - /solutions/storage_latest.iso
+      - /other_solutions/computing.iso
+
+There would be no explicit information about what the archives contain.
+Instead, we want the archive itself to contain such information (more
+details in :doc:`./archive`), and to discover it at import time.
+
+Responsibilities of each party
+------------------------------
+
+This section intends to define the boundaries between MetalK8s and the
+Solutions to integrate with, in terms of "who is doing what?".
+
+.. note:: This is still a work in progress.
+
+MetalK8s
+^^^^^^^^
+
+**MUST**:
+
+- Handle reading and mounting of the Solution ISO archive
+- Provide tooling to deploy/upgrade a Solution's CRDs and UI
+
+**MAY**:
+
+- Provide tooling to deploy/upgrade a Solution's Operator
+- Provide tooling to verify signatures in a Solution ISO
+- Expose management of Solutions in its own UI
+
+Solution
+^^^^^^^^
+
+**MUST**:
+
+- Comply with the standard archive structure defined by MetalK8s
+- Expose management of its Operator instances in its UI
+- Handle monitoring of its own services (both Operator and application, except
+  the UI)
+
+**MAY**:
+
+- Use MetalK8s monitoring services (Prometheus and Grafana)
```

Comment:
I thought there was no strong requirement, so moved it to _MAY_, but indeed it makes sense to force the usage of the cluster-provided Prometheus stack.

Reply:
I don't have a strong opinion either way, it's mostly about nudging developers in a "preferred" direction so that resources/storage capacity planning for the platform's Prometheus instance are adequate. As a Solution developer I would enjoy not having to manage my own Prom stack if I can trust that I won't have any surprises with retention duration etc :)

Reply:
Changed `MAY` into `SHOULD`, although I don't want to say `MUST`, since deploying your own monitoring stack should not prevent the Solution from working (though I'm not sure if having multiple Prometheus Operators watching for `ServiceMonitor`s could pose issues...). I'll open a tracking ticket.

Reply:
Ok, latest conversations led to a simpler stance: Solutions that want to be deployed on MetalK8s `MUST` use the provided Prometheus Operator to set-up their monitoring. However, it is reasonable to assume Solution developers may want their Operator to work in any K8s cluster, hence could manage their own monitoring stack there. I'm still unsure about how they would do so, let's focus on _using_ MetalK8s stack for now (will consider the alternative use-case in #1158).